source/objects/simulationClass.m

```diff
@@ -28,40 +28,40 @@
+            'endTime',      100, ...                                    % 
+            'dt',           0.1, ...                                    %             
+            'path',         'vtk')                                      % (`structure`) Defines the BEM data implemtation. ``option`` (`integer`) Flag for paraview visualization, and writing vtp files, Options: 0 (off) , 1 (on). Default = ``0``. ``startTime`` (`float`) Start time for the vtk file of Paraview. Default = ``0``. ``endTime`` (`float`) End time for the vtk file of Paraview. Default = ``100``.  ``dt`` (`float`) Timestep for Paraview. Default = ``0.1``. ``path`` (`string`) Path of the folder for Paraview vtk files. Default = ``'vtk'``.        
+        pressure {mustBeMember(pressure,0:1)}         = 0               % (`integer`) Flag to save pressure distribution, Options: 0 (off), 1 (on). Default = ``0``
+        rampTime {mustBeNonnegative}                  = 100             % (`float`) Ramp time for wave forcing. Default = ``100`` s        
+        rateTransition                                = 'on'            % (`string`) Flag for automatically handling rate transition for data transfer, Opyions: 'on', 'off'. Default = ``'on'``
+        reloadH5Data {mustBeMember(reloadH5Data,0:1)} = 0               % (`integer`) Flag to re-load hydro data from h5 file between runs, Options: 0 (off), 1 (on). Default = ``0``
+        rho {mustBePositive}                          = 1000            % (`float`) Density of water. Default = ``1000`` kg/m^3
+        saveStructure {mustBeMember(saveStructure,0:1)} = 0             % (`integer`) Flag to save results as a MATLAB structure, Options: 0 (off), 1 (on). Default = ``1``
+        saveText {mustBeMember(saveText,0:1)}         = 0               % (`integer`) Flag to save results as ASCII files, Options: 0 (off), 1 (on). Default = ``0``
+        saveWorkspace {mustBeMember(saveWorkspace,0:1)} = 1             % (`integer`) FLag to save .mat file for each run, Options: 0 (off), 1 (on). Default = ``1``
+        simMechanicsFile                              = 'NOT DEFINED'   % (`string`) Simulink/SimMechanics model file. Default = ``'NOT DEFINED'``
+        solver                                        = 'ode4'          % (`string`) PDE solver used by the Simulink/SimMechanics simulation. Any continuous solver in Simulink possible. Recommended to use 'ode4, 'ode45' for WEC-Sim. Default = ``'ode4'``
+        stateSpace {mustBeMember(stateSpace,0:1)}     = 0               % (`integer`) Flag for convolution integral or state-space calculation, Options: 0 (convolution integral), 1 (state-space). Default = ``0``
+        startTime {mustBeNonnegative}                 = 0               % (`float`) Simulation start time. Default = ``0`` s        
+        zeroCross                                     = 'DisableAll'    % (`string`) Disable zero cross control. Default = ``'DisableAll'``

```

Comment:
How do these property assets work? Can you share a MATLAB link? How does it affect our documentation, etc? Are you planning to add this to the other classes too?

Reply:
@jtgrasb I like this feature a lot, and think we should implement it, but I'm wondering whether it's something we should push to after v5.0 and implement across all the classes, or whether we should work to include it in v5.0, LMK what you think.

Reply:
@kmruehl I agree, it is a very useful feature. I think it may be best to wait until after v5.0 because although helpful, I don't think it affects WEC-Sim significantly with most use cases. It would also give me more time to work out the kinks and implement it for the other classes. If you think it should be part of v5.0 though, I can try to get the rest of the classes implemented.

Reply:
I agree, let's hold off of this on v5.0, but please add it to the project board for future development.

Reply:
Actually, can you revert the simulationClass property validation, and then I'll merge this.